sdk/core/azure-core/azure/core/tracing/_inference_api_instrumentor_impl.py

```diff
@@ -0,0 +1,379 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import asyncio
+import functools
+import importlib
+import json
+import logging
+from enum import Enum
+from typing import Any, Iterator, Callable, Optional, List
+from azure.ai.inference.aio import ChatCompletionsClient
+from azure.ai.inference import models as _models
+from azure.core.tracing import AbstractSpan
+from azure.core.tracing import SpanKind
+from azure.core.settings import settings
+from .common import get_function_and_class_name
+
+_inference_traces_enabled: bool = False
+_trace_inference_content: bool = False
+INFERENCE_GEN_AI_SYSTEM_NAME = "azure.ai.inference"
+
+class TraceType(str, Enum):
+    """An enumeration class to represent different types of traces."""
+
+    INFERENCE = "Inference"
+
+
+def _set_attributes(span: AbstractSpan, *attrs: tuple[str, Any]) -> None:
+    for attr in attrs:
+        key, value = attr
+        if value is not None:
+            span.add_attribute(key, value)
+
+
+def _add_request_chat_message_event(span: AbstractSpan, **kwargs: Any) -> None:
+    for message in kwargs.get("messages", []):
+        try:
+            message = message.as_dict()
+        except AttributeError:
+            pass
+
+        if message.get("role"):
+            name = f"gen_ai.{message.get('role')}.message"
+            span.span_instance.add_event(
+                name=name,
+                attributes={
+                    "get_ai.system": INFERENCE_GEN_AI_SYSTEM_NAME,
+                    "gen_ai.event.content": json.dumps(message)
+                }
+            )
+
+
+def _add_request_chat_attributes(span: AbstractSpan, **kwargs: Any) -> None:
+    _set_attributes(
+        span,
+        ("gen_ai.system", INFERENCE_GEN_AI_SYSTEM_NAME),
```

Comment:
we also need `gen_ai.operation.name` (only `chat` and legacy completions is defined for now, lmk if you want to add some more)

Reply:
added